### PR TITLE
feat(modal): add fade animation

### PR DESCRIPTION
**Pull request by jasonwoodland**
**What this PR does** 📖
- Add fade in/out animation to modal

**Which issue(s) this PR fixes** 🔨
AP-1917

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

File: assets/styles/framework/animations.less [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Animations
 @animation-speed: 0.05s;
 @animation-speed-micro: 0.05s;
+@animation-speed-slow: 0.25s;
```
Comment (josephmcg) [inline assets/styles/framework/animations.less:4]: could you clean these up while you're working on it? maybe remove micro and switch all instances of that to animation-speed

File: components/ui/Modal/Modal.less [hunk 2]
```diff
@@ -92,3 +94,25 @@
...
+  .modal-background {
+    background-color: transparent !important;
+    backdrop-filter: blur(0px) !important;
+  }
+
+  .modal-card {
+    opacity: 0 !important;
+    transform: perspective(400px) rotateX(2deg) !important;
+    transform-origin: 50% 200%;
+    filter: @light-blur !important;
```
Comment (josephmcg) [inline components/ui/Modal/Modal.less:116]: any way to avoid important? if this is overriding other important, try removing those instead

**Review by josephmcg — CHANGES_REQUESTED** [review]

File: assets/styles/framework/animations.less (hunk 1 above)
Comment (jasonwoodland) [reply assets/styles/framework/animations.less:4]: you mean consolidate all mentions of `@animation-speed-micro` to just `@animation-speed`?

I know they're the same value now, but maybe we might want to differentiate in the future? Or is it safe to do so?

File: components/ui/Modal/Modal.less (hunk 2 above)
Comment (jasonwoodland) [reply components/ui/Modal/Modal.less:116]: oops, forgot to remove that 😅

File: assets/styles/framework/animations.less (hunk 1 above)
Comment (josephmcg) [reply assets/styles/framework/animations.less:4]: Yep, thats what i mean. If theyre the same value, they might as well use the same variable. If somebody wants to add a third speed later, they can

**Review by josephmcg — APPROVED** [review]

**Comment by phillsatellite** [comment]
Found small UI issue when navigating from friends page to the Coming Soon modal looks like you can see the modal sliding in
https://user-images.githubusercontent.com/93608357/176947139-c53c0282-449c-4522-9173-f7562dbc8fa9.mp4

**Comment by jasonwoodland** [comment]
@phillsatellite good spot, was this on safari? I think it was animating the transform-origin value which caused the glitch. Anyway I've moved the transform-origin outside of the animation ruleset.

**Comment by phillsatellite** [comment]
> @phillsatellite good spot, was this on safari? I think it was animating the transform-origin value which caused the glitch. Anyway I've moved the transform-origin outside of the animation ruleset.

Hey @jasonwoodland! It was on Safari 🔥

**Review by josephmcg — APPROVED** [review]

**Pull request merged**